lib/puppet/type/sensuctl_config.rb

```diff
@@ -0,0 +1,37 @@
+require_relative '../../puppet_x/sensu/type'
+require_relative '../../puppet_x/sensu/array_property'
+require_relative '../../puppet_x/sensu/hash_property'
+require_relative '../../puppet_x/sensu/integer_property'
+
+#Dir[File.dirname(__FILE__) + '/sensu_*.rb'].each do |file|
+#  next if file == __FILE__
+#  next if File.basename(file) == 'sensu_api_validator.rb'
+#  require file
+#end
+
+Puppet::Type.newtype(:sensuctl_config) do
+  desc <<-DESC
+@summary Abstract type to configure other types
+DESC
+
+  newparam(:name, :namevar => true) do
+    desc "The name of the resource."
+  end
+
+  newparam(:chunk_size, :parent => PuppetX::Sensu::IntegerProperty) do
+    desc "sensuctl chunk-size"
+  end
+
+  def generate
```

Comment:
i dont follow how this works

Reply:
The `generate` function I think is intended to allow a type to define other types or in this case modify other types. What we end up doing here is setting class variable `chunk_size` for all types.  That is used when `sensuctl <resource> list` is executed.

Reply:
The logic on looking at files is listing all providers that use `sensuctl` since those are the ones we are modifying the class variable for.  This is one approach to make things dynamic, other modules I've seen use `generate` have just hardcoded the list of types that should be modified.

Reply:
since this is not standard to how the rest of the params work, could you document this in the code.

Reply:
Added comments to code